packages/experimental/Icon/src/Icon.tsx

```diff
@@ -65,16 +65,20 @@ function renderSvg(iconProps: IconProps) {
     ? '#FFFFFF'
     : '#000000';
 
+  // The svg color can be set using either style.color or iconProps.color, where style.color is preferred in case both are set.
+  const colorString = (style as any).color !== undefined ? (style as any).color : iconColor;
+  const color = (Platform.OS == 'macos' ? processColor(colorString as ColorValue) : colorString) as Color;
```

Comment:
I totally missed this earlier, but I have found that calling processColor directly is a bug. Depending on the order of calls, it can swap RGBA bits into ARGB, which renders the wrong color. also, of the reason we're doing this only for macOS is because macOS uses PlatformColor, converting to a string won't do anything since platformColors cannot be represented as strings. We should make a follow-up here to remove this logic.

Reply:
I think we made sure that the iconProps color is always a string until we support ColorValue objects on rn-svg. I can find the PR / code where me made that change if we're curious. 
But basically you should never pass an SVG a ColorValue for a color because ColorValue could be an object like a PlatformColor. Instead, pass it the icon color bc we made sure that should be a string I think.

Reply:
No, RN-SVG doesn't support ColorValue at all because it predates RN 0.63 and hasn't been updated to support it yet. 
@rurikoaraki and I talked about it some when she added support for PlatformColor in our theme definition, and we made sure to keep tokens that apply to SVGs (basically Icons) to only be strings and not ColorValues.

A followup (for me) is to update RN-SVG to support ColorValue. There's an [open PR](https://github.com/react-native-svg/react-native-svg/pull/1561) that adds support for PlatformColor, but it's not very extensible to out of tree platforms